packages/castmill/lib/castmill_web/channels/device_updates_channel.ex

```diff
@@ -0,0 +1,54 @@
+defmodule CastmillWeb.DeviceUpdatesChannel do
+  use CastmillWeb, :channel
+  alias Castmill.Devices
+  alias Castmill.Organizations
+
+  require Logger
+
+  @impl true
+  def join("device_updates:" <> device_id, _params, socket) do
+    %{:user => user} = socket.assigns
+
+    Logger.info("User #{user.id} is joining device_updates channel for device #{device_id}")
+
+    if authorized?(user.id, device_id) do
+      {:ok, assign(socket, :device_id, device_id)}
+    else
+      {:error, %{reason: "unauthorized"}}
+    end
+  end
+
+  # Handle all messages that are not handled by the
+  # `handle_in/3` callbacks above
+  @impl true
+  def handle_in(event, payload, socket) do
+    IO.puts("Unhandled event #{inspect(event)} #{inspect(payload)}")
+    {:noreply, socket}
```

Comment:
Are these default implementationes, ment to be overridden?

Reply:
handle_in handles messages from the client to the server